packages/datadog-plugin-mongodb-core/src/index.js

```diff
@@ -69,12 +69,25 @@ function getResource (ns, query, operationName) {
   return parts.join(' ')
 }
 
+function truncate (input) {
+  return input.slice(0, Math.min(input.length, 10000))
+}
+
+function simplify (input) {
+  return isBSON(input) ? input.toHexString() : input
+}
+
+function shouldSimplify (input) {
+  return !isObject(input) || isBSON(input)
+}
+
 function shouldHide (input) {
-  return !isObject(input) || Buffer.isBuffer(input) || isBSON(input)
+  return Buffer.isBuffer(input) || typeof input === 'function'
```

Comment:
Functions should probably just be removed entirely.

Reply:
For buffers, since now we want to actually send values, they should be converted to a string if possible.

Reply:
Buffers could be raw binary data though. Would it be safe to stringify that?

Reply:
Any way we can know if it's a string? When testing this I was getting some strings as buffers, so it seems to be fairly standard.

Reply:
No easy way, no. Buffers make no assertions about what encoding it contains, leaving it up to the user to _tell_ it what encoding it contains when they try to stringify it. We can't really infer that automatically. I don't see a viable path forward unless we analyze the full contents of every buffer which would likely be very expensive and error prone.

Reply:
Ok let's keep it out for now then and address it if it's problematic for users.